internal/middleware/middleware.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"eventapi/internal/session"
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/redis/go-redis/v9"
+)
+
+func UseSession(s session.SessionManager) func(*fiber.Ctx) error {
+	return func(c *fiber.Ctx) error {
+		sId := c.Cookies("session")
+		if sId != "" {
+			sessionData, err := s.Get(c, sId)
+			if err == redis.Nil {
+				s.Clear(c, sId)
+				return c.Redirect("/api/login", 303)
+			}
```

Comment:
my first thought was just to use redis but since I am going to host this somewhere I like the idea of simplifying for now since I dont really need redis right now.